frameworks/fmi2api/src/main/java/org/intocps/maestro/framework/fmi2/api/mabl/scoping/ScopeFmi2Api.java

```diff
@@ -219,6 +216,92 @@ protected StringVariableFmi2Api store(Supplier<String> nameProvider, String valu
                 newAIdentifierExp(name));
     }
 
+    /**
+     *
+     * @param identifyingName the name of the array
+     * @param mdArray non-jagged multidimensional array.
+     * @param <V> data type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArray(String identifyingName, V[] mdArray) {
+        List<Integer> arrayShape = new ArrayList<>();
+        PType type;
+        V[] subArr = mdArray;
+        while(subArr.getClass().getComponentType().isArray()){
+            arrayShape.add(subArr.length);
+            subArr = (V[])subArr[0];
+        }
+        arrayShape.add(subArr.length);
+
+        if (subArr instanceof Double[]) {
+            type = newARealNumericPrimitiveType();
+        } else if (subArr instanceof Integer[]) {
+            type = newAIntNumericPrimitiveType();
+        } else if (subArr instanceof Boolean[]) {
+            type = newABoleanPrimitiveType();
+        } else if (subArr instanceof String[]) {
+            type = newAStringPrimitiveType();
+        } else { throw new IllegalArgumentException();  }
+
+        PStm arrayVariableStm = newALocalVariableStm(
+                newAVariableDeclarationMultiDimensionalArray(newAIdentifier(identifyingName), type, arrayShape));
+
+        add(arrayVariableStm);
+
+        return createMDArrayRecursively(mdArray, arrayVariableStm,
+                newAIdentifierStateDesignator(newAIdentifier(identifyingName)), newAIdentifierExp(identifyingName));
+    }
+
+    /**
+     * @param array          multi dimensional array
+     * @param declaringStm   declaring statement of the outer array
+     * @param <V>            generic type
+     * @return an ArrayVariable representing the multidimensional array
+     */
+    private <V> ArrayVariableFmi2Api<V> createMDArrayRecursively(V[] array, PStm declaringStm, PStateDesignatorBase stateDesignator, PExpBase indexExp) {
```

Comment:
What does this function do that differs from the create? 
it looks like it should just be an other overload of the `createArray` or more likely a version of `store...`

Reply:
This function is made to handle the recursive loop to create the array structure after the declaring statement has been created in the function 'createMDArray'. These could possible be merged into one function.
I'm unsure how to overload the store function to handle multi dimensional arrays but im probably missing or misunderstanding something, so if you have time for a meeting I would like to hear your thoughts.